thunder/transforms/quantization.py

```diff
@@ -294,3 +295,214 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+                with tracectx(prologue_trace):
+                    # better way
+                    proxy_loraA = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_loraA",
+                        shape=qs["loraA.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["loraA.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    proxy_loraB = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_loraB",
+                        shape=qs["loraB.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["loraB.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_loraA, output=proxy_loraA))
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_loraB, output=proxy_loraB))
+
+                    add_trace_output(prologue_trace, proxy_loraA, subindex=0)
+                    add_trace_output(prologue_trace, proxy_loraB, subindex=0)
+
+                    new_compute_inputs.append(proxy_loraA)
+                    new_compute_inputs.append(proxy_loraB)
+                    # this is not good, because we will have several traces...
+                    additional_proxies[n_loraA] = proxy_loraA
+                    additional_proxies[n_loraB] = proxy_loraB
+
+        prologue_trace.bound_symbols[-1:-1] = new_bsyms
+
+        computation_proxy_map = {
+            csym.name: dict(
+                shape=psym.shape,
+                dtype=psym.dtype,
+            )
+            for psym, csym in zip(prologue_trace.bound_symbols[-1].args[0][0], computation_trace.args)
+            if psym.shape != csym.shape or psym.dtype != csym.dtype
+        }
+
+        new_computation_trace = trace_with_replaced_proxy_metadata(computation_trace, computation_proxy_map)
+        bound_symbols = new_computation_trace.bound_symbols
+        new_computation_trace.bound_symbols = []
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear:
+                n = lora_linear_proxies[bsym.args[1].name]
```

Comment:
I think we need to check whether the name is in the dict, in case people there are other linears.